base/src/main/java/io/spine/query/package-info.java

```diff
@@ -19,7 +19,386 @@
+ * correspond to the types of the declared columns. That is, {@code whenFounded} column would
+ * only accept {@code Timestamp} values for comparison, and {@code isTraded} only
+ * accepts {@code Boolean}s.
+ *
+ * <p>It is also possible to build a query with a disjunction (OR) operator:
+ *
+ * <pre>
+ *     {@literal RecordQuery<ManufacturerId, Manufacturer> query = }
+ *                     manufacturerBuilder()
+ *                             .either((b) -> b.where(whenFounded).isLessThan(firstJanuary1990),
+ *                                     (b) -> b.where(isTraded).is(true))
+ *                             .build();
+ * </pre>
+ *
+ * <p>Where {@code b} is the instance of the same {@code RecordQueryBuilder}, for proper chaining.
+ * This query selects all the records which have either {@code whenFounded} value less than
+ * January, 1, 1990, or the {@code isTraded} value equal to {@code true}.
+ *
+ * <p>Please note that {@code ManufacturerId} is used as a type of the record identifier when
+ * creating a builder, as it is the first field declared in this message. Its values may
+ * also be used in a query builder:
+ *
+ * <pre>
+ *    {@literal RecordQuery<ManufacturerId, Manufacturer> query = }
+ *        RecordQuery.newBuilder(ManufacturerId.class, Manufacturer.class)
+ *                   .id().is(someManufacturer)
+ *                   .build();
+ * </pre>
+ *
+ * <p>Where {@code someManufacturer} must a value of {@code ManufacturerId} type. In addition
+ * to {@link io.spine.query.IdCriterion#is(java.lang.Object) is(value)}, the framework also provides
+ * {@link io.spine.query.IdCriterion#in(java.lang.Object[]) in(...values)} and
+ * {@link io.spine.query.IdCriterion#in(java.lang.Iterable) in(Iterable values)}
+ * overloads. They select the records by several ID values.
+ *
+ * <h4>Ordering and limit</h4>
+ *
+ * <p>The query builder API also allows specifying the ordering and record limit for the queries:
+ *
+ * <pre>
+ *     {@literal RecordQuery<ManufacturerId, Manufacturer> query = }
+ *     {@literal    RecordQuery.newBuilder(ManufacturerId.class, Manufacturer.class) }
+ *                    .orderBy(whenFounded, ASC)
+ *                    .orderBy(isTraded, DESC)
+ *                    .limit(10)
+ *                    .build();
+ * </pre>
+ *
+ * <p>This query selects all records of {@code Manufacturer} type, but tells to order them
+ * by the values of {@code whenFounded} column ascending, then by {@code isTraded} column values
+ * descending. Finally, the query result is limited to the top ten records.
+ *
+ * <p>Please note, that {@linkplain io.spine.query.RecordQueryBuilder#limit(int) limit(..)}
+ * method may only be used in queries which order their results. Otherwise, a runtime exception
+ * is thrown upon building the query.
+ *
+ * <h4>Field masks</h4>
+ *
+ * <p>Sometimes end-users would want to fetch only the particular fields of the stored messages.
```

Comment:
I'm not sure about “the particular fields”. We have not spoke about these fields yet, so the definite article may not apply.